lib/req/transport_error.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Req.TransportError do
+  @moduledoc """
+  Represents an error with the transport used by an HTTP connection.
+
+  This is a standardised exception that all Req adapters should use for transport layer related
```

Comment:
I’m a nitpicker at heart

```suggestion
  This is a standardised exception that all Req adapters should use for transport-layer-related
```